ui/src/router.tsx

```diff
@@ -42,12 +47,18 @@ export function AppRouter() {
   );
 }
 
+// TODO(tjennison): This is becoming spaghetti. Consider alternative ways to set
+// this up or perhaps alternative libraries.
 export function underlayURL(underlayName: string) {
   return underlayName;
 }
 
-export function cohortURL(cohortId: string, groupId: string) {
-  return "cohorts/" + cohortId + "/" + groupId;
+export function cohortURL(cohortId: string, groupId?: string) {
```

Comment:
Is this function called from somewhere other than `absoluteCohortURL` below? If not, maybe you can inline it there instead.